executor/load_data.go

```diff
@@ -180,17 +180,74 @@ func (e *LoadDataExec) loadFromRemote(
 	if err != nil {
 		return ErrLoadDataCantAccess
 	}
-	fileReader, err := s.Open(ctx, filename)
-	if err != nil {
-		return ErrLoadDataCantRead.GenWithStackByArgs(getMsgFromBRError(err), "Please check the INFILE path is correct")
+
+	idx := strings.IndexByte(path, '*')
+	// simple path when the INFILE represent one file
+	if idx == -1 {
+		opener := func(ctx context.Context) (io.ReadSeekCloser, error) {
+			fileReader, err2 := s.Open(ctx, path)
+			if err2 != nil {
+				return nil, ErrLoadDataCantRead.GenWithStackByArgs(getMsgFromBRError(err2), "Please check the INFILE path is correct")
+			}
+			return fileReader, nil
+		}
+
+		filesize := int64(-1)
+		reader, err2 := opener(ctx)
+		if err2 == nil {
```

Comment:
odd to continue load when open failed, we don't have retry for this now, maybe just return err